### PR TITLE
data: add demo subjects vocabulary

### LINKED ISSUE #168: data: add demo subjects
Subjects vocabulary are quite instance specific. However, as an example (and to facilite it's customization), we can create a demo one (e.g. `science`: cosmology, astronomy, biology...)

### PR DESCRIPTION
**Pull request by ppanero**
- closes #168 
- closes https://github.com/inveniosoftware/cookiecutter-invenio-rdm/issues/171

### CODE REVIEW

**Comment by ntarocco** [comment]
Can we move this code to the `demo` instance instead? So that we can have an example there?

File: {{cookiecutter.project_shortname}}/app_data/vocabularies/subjects.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+- id: particle
+   props:
```
Comment (ntarocco) [inline {{cookiecutter.project_shortname}}/app_data/vocabularies/subjects.yaml:2]: extra space everywhere?

**Comment by ntarocco** [comment]
I was testing subjects locally, copy/pasted and saw the extra space.

File: {{cookiecutter.project_shortname}}/app_data/vocabularies/subjects_default.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+- id: "https://inveniosoftware.org/products/rdm/subjects/default/1"
```
Comment (fenekku) [inline {{cookiecutter.project_shortname}}/app_data/vocabularies/subjects_default.yaml:1]: Using this non-resolving URI as a way to get a unique URI ID

File: {{cookiecutter.project_shortname}}/app_data/vocabularies/subjects.yaml (hunk 1 above)
Comment (ppanero) [reply {{cookiecutter.project_shortname}}/app_data/vocabularies/subjects.yaml]: fixed :)

**Comment by ppanero** [comment]
Screenshot 2021-07-16 at 10 26 08

**Pull request merged**